pkgs/tools/filesystems/dwarfs/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, bison
+, boost
+, cmake
+, double-conversion
+, fetchFromGitHub
+, fmt_8
+, fuse3
+, gflags
+, git
+, glog
+, jemalloc
+, lib
+, libarchive
+, libdwarf
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkgconfig
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # Needs tagged git repo for generating some files
+  # leaveDotGit alone isn't enough and breaks reproducible hashing
+  preConfigure = ''
+    git init
+    git add .
+    git -c user.name=root -c user.email=root@localhost commit -m "v${version}" -q
+    git -c user.name=root -c user.email=root@localhost tag "v${version}" -m "v${version}"
+  '';
+
+  # ...but the files are still kinda ugly, so let's make the data prettier for wherever it gets used ;)
+  postConfigure = ''
+    sed -i \
+      -e 's/\(PRJ_GIT_REV = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_DESC = \).*/\1"${version}";/' \
+      -e 's/\(PRJ_GIT_BRANCH = \).*/\1"${src.rev}";/' \
+      -e 's/\(PRJ_GIT_ID = \).*/\1"${version} from Nixpkgs :)";/' \
+      ../src/dwarfs/version.cpp
+  '';
+
+  cmakeFlags = [
+    "-DPREFER_SYSTEM_ZSTD=ON"
+    "-DPREFER_SYSTEM_XXHASH=ON"
+  ];
+
+  nativeBuildInputs = [
+    bison
+    boost.dev
+    cmake
+    fmt_8
+    git
+    libdwarf.dev
```

Comment:
`libdwarf` was recently marked insecure which prevents this package from being built automatically. Furthermore, dwarfs seems to build fine without it. I suggest removing it, unless something in particular is broken?

Reply:
Interestingly `libdwarf` is marked as a dependency of `folly` (Facebook's dependency resolution script doesn't seem to want to output the dependencies and I can't figure out where to look for them in the source, so I'm looking at [the AUR](https://aur.archlinux.org/packages/folly)). We're also missing some dependencies, although it seems to compile just fine. I'm not entirely sure on how this works.

Reply:
I usually start with nothing and add dependencies until it works. It seems `libdwarf` is only used for an experimental feature `dwarfs` has no need for so I don't see a reason to include it. Nix's default Folly build does not include it either.